### PR TITLE
Save request chunks for handlers that wish to pipe complete requests to other servers

### PR DESCRIPTION
**Pull request by jankeromnes**
What the title and 8f8204aa850533a3f3bd621339e4849b4153fce1 say.

Warning: This pull request is based on #45, which will need to be merged first before this one makes sense.

### CODE REVIEW

**Comment by jankeromnes** [comment]
Note: To preview changes from this pull request only (without the unrelated commits from #45) please view the "Add a server option to save processed request chunks" commit by itself: https://github.com/espadrine/sc/pull/60/commits/8f8204aa850533a3f3bd621339e4849b4153fce1

**Comment by jankeromnes** [comment]
The rebased commit for this pull request is now https://github.com/espadrine/sc/pull/60/commits/34a8855d4ceeb9c568e7571b30dea8f5717f1033 (the rest is still from #45).

**Comment by jankeromnes** [comment]
Rebased again, and ready for review! With a couple drive-by improvements for free.

Please have a look whenever it's convenient :)

File: test/test-api.js [hunk 1]
```diff
@@ -173,6 +173,38 @@ var launchTests = function () {
           });
         });
       });
+    },
+
+    function t5 (next) {
+      var name = 'Robert\'); DROP TABLE Students;--';
```
Comment (espadrine) [inline test/test-api.js]: niiice

**Comment by espadrine** [comment]
Thanks a lot for the contribution!

**Comment by jankeromnes** [comment]
Thanks a lot for the quick merge! 🎆

**Pull request merged**